src/prometheus/prometheus.c

```diff
@@ -120,15 +121,20 @@ static int prometheus_init(void) {
     curl_easy_setopt(curl, CURLOPT_USERNAME, user);
     curl_easy_setopt(curl, CURLOPT_PASSWORD, (pass == NULL) ? "" : pass);
 #else
-    static char credentials[1024];
-    int status = ssnprintf(credentials, sizeof(credentials), "%s:%s", user,
+    size_t credentials_size = strlen(user) + sizeof(":") + strlen(pass);
```

Comment:
I think that `sizeof(":")` would return 8 since `":"` is a `const char *` and size of the pointer is 8. Is it what you what?

Reply:
Because it's a constant, at least GCC treats it same as:
`const char dcolon[] = ":";`
Which `sizeof()` is 2.

Reply:
When optimizations are enabled, the code generated for `2`, `sizeof(":")` and `strlen(":")+1"` are all identical.  Using just `strlen(":")` would IMHO be clearest though.